### PR TITLE
[SOL] Change encoding of `callx` instruction

### PR DESCRIPTION
**Pull request by LucasSte**
This PR is one of the tasks in https://github.com/solana-labs/solana/issues/34250 to update the SBF target with the new instructions. Especially, this modification is the equivalent of https://github.com/solana-labs/rbpf/pull/491 in LLVM, changing the encoding of the `callx` instruction.

There was a small refactoring in the table gen files to encompass a new `SubtargetFeature`. As `FeatureCallxRegSrc` was required both in `SBF.td` and in `SBFInstrInfo.td`, I put all the feature in a new file to avoid a circular dependency.

### CODE REVIEW

File: llvm/test/MC/SBF/sbf-jmp.s
Comment (dmakarov) [inline llvm/test/MC/SBF/sbf-jmp.s]: maybe this won't be necessary when you rebase on #77?

**Review by dmakarov — COMMENTED** [review]
LGTM, but it's still a Draft. Are planning more changes in this PR?

**Comment by LucasSte** [comment]
I tested this change against the programs in `programs/sbf` on the Solana's main repository. Besides the instruction count that has slightly changed (we now have two instructions for `lddw`: `mov32` and `hor64`), all tests have passed.

I'm waiting to merge #77 so that I can rebase this PR and fix the conflicts before the review.

File: llvm/test/MC/SBF/sbf-jmp.s
Comment (LucasSte) [reply llvm/test/MC/SBF/sbf-jmp.s]: I'll remove it once I merge that.

Comment (dmakarov) [reply llvm/test/MC/SBF/sbf-jmp.s]: Thanks

**Comment by LucasSte** [comment]
> LGTM, but it's still a Draft. Are planning more changes in this PR?

I converted to a draft because I wanted to test it with real programs first. I also noticed I got the encoding wrong 😅

**Comment by dmakarov** [comment]
LGTM, please feel free to merge when the CI is done.

**Comment by LucasSte** [comment]
> LGTM, please feel free to merge when the CI is done.

Merging is blocked without an approval.

**Review by dmakarov — APPROVED** [review]
Sorry, forget to approve.

**Pull request merged**
